src/Compilers/Core/Portable/Syntax/InternalSyntaxNodeExtensions.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System;
+
+namespace Microsoft.CodeAnalysis
+{
+    internal static class InternalSyntaxNodeExtensions
```

Comment:
>InternalSyntaxNodeExtensions [](start = 26, length = 28)

I am not sure what the use of "Internal" in the name is supposed to convey. Is this about accessibility of the helpers inside the type? If so, I think we should put them into existing type and into the src\Compilers\Core\Portable\Syntax\SyntaxNodeExtensions.cs file.

Reply:
So the new API should be public?

Reply:
We need this API in IDE layers. That leaves three options:

1. Make the new code a public API
2. Make the new code live in a new file, and link that file into the workspaces layer (current approach)
3. Make the code an internal API in SyntaxNodeExtensions.cs, and duplicate the code to a different file for use in the IDE layers

Option 1 or 2 is the most desirable for reducing duplication. Option 2 has no public API review requirement. Any of the three would work though.

Reply:
> So the new API should be public?

I would be fine with that. Also, it looks like we already have an API with this name as in instance method on SyntaxNode. We probably should place the new API next to it as an overload.

---
In reply to: [400949996](https://github.com/dotnet/roslyn/pull/42893#discussion_r400949996) [](ancestors = 400949996)